lib/matplotlib/axes/_base.py

```diff
@@ -1680,6 +1680,8 @@ def set_aspect(self, aspect, adjustable=None, anchor=None, share=False):
             aspect = 1
         if not cbook._str_equal(aspect, 'auto'):
             aspect = float(aspect)  # raise ValueError if necessary
+            if aspect <= 0 or ~np.isfinite(aspect):
```

Comment:
I suspect `~` cannot be used here (and even if it could, we should use the standard `not` operator.
Note that `aspect` is scalar and thus `np.isfinite(aspect)` as a bool, not a numpy array. The standard negation for bools is `not`.

I also suggest to write it this way, which is the direct negation of the requirement stated in the error message:
```suggestion
            if not (np.isfinite(aspect) and aspect > 0):
```

Reply:
```ipython
In [4]: aspect = np.Inf

In [5]: print(aspect <= 0 or ~np.isfinite(aspect))
True
```

I personally find `if not (A and B)` harder to read than `if not(A) or not(B)`, but your mileage may vary.

Reply:
Ah, the trick is that `np.isfinite` returns `np.bool_` and not `bool` (even though the documentation says `bool`

```
In [3]: type(np.isfinite(0))
Out[3]: numpy.bool_
```

Anyway, I propose not to rely on that detail, because the bitwise negation on `bool` is funky and could confuse readers:

```
In [4]: ~True
Out[4]: -2

In [5]: ~False
Out[5]: -1
```

I'm also ok with
```suggestion
            if not np.isfinite(aspect) or aspect <= 0:
```